### PR TITLE
solve blocking weaponization

### LINKED ISSUE #175: Solve the Blocking Weaponization Issue
Right now the block page says "They cannot reply to your content or notify you with a username mention". I don't know if this is accurate. Check to see if it's accurate!

If it is, I think we should change it to eliminate notifications, both username mentions and reply mentions, and maybe to hide the blockee from the blocker's view. This makes it less of a "block" and more of a "hide" option, but that's, y'know, kind of what we want. But don't actually restrict posting - you shouldn't be able to prevent someone from posting.

If you have a better idea, though, let me know.

I think it is unlikely that we run into trouble with this immediately but it should probably be fixed at some point.

### PR DESCRIPTION
**Pull request by justcool393**
fixes #175

### CODE REVIEW

**Comment by justcool393** [comment]
this should be it but it needs testing

File: files/routes/comments.py
Comment (TLSM) [inline files/routes/comments.py]: `comment_on_publish` is used in at least one other location (admin.py for approving filtered comments) so if the signature is going to change, you have to propagate it there too. However, I think it might be better to move replying_to_blocked into the User model somehow and just call it separately inside comment_on_publish and the api_comment route.

File: files/templates/settings_blocks.html
Comment (TLSM) [inline files/templates/settings_blocks.html]: It also means a blocker doesn't see other sources of notifications from their blockee (namely reply notifs) and also their blockee's comments are hidden.

**Review by TLSM — CHANGES_REQUESTED** [review]
The change to User.userblocks looks to correctly make blocks more unidirectional: blocks no longer prevent the blockee from seeing the blocker's content or replying to them, while still allowing the blocker to not be notified of nor otherwise see the blockee.

Change requested to fix a bug resulting from changing comment_on_publish signature, also clarifying some UI text.

**Review by TLSM — APPROVED** [review]
Tested and requested changes work correctly on happy paths for new reply comment and filtered comment approval. Probably still some untested edge cases, but it at least won't violently break prod if so.

**Comment by zorbathut** [comment]
Looks good!

(1) Something I checked in before is conflicting, sorry (I suspect it's the big error change.)

(2) I'm gonna make a meta post asking about this before I actually merge this in, though I doubt there will be major problems.

**Pull request merged**
